src/matrix-utils.ts

```diff
@@ -1,34 +1,49 @@
-import matrix from "matrix-js-sdk/src/browser-index";
+import Olm from "@matrix-org/olm";
+import olmWasmPath from "@matrix-org/olm/olm.wasm?url";
+import { IndexedDBStore } from "matrix-js-sdk/src/store/indexeddb";
+import { WebStorageSessionStore } from "matrix-js-sdk/src/store/session/webstorage";
+import { MemoryStore } from "matrix-js-sdk/src/store/memory";
+import { IndexedDBCryptoStore } from "matrix-js-sdk/src/crypto/store/indexeddb-crypto-store";
+import { createClient, MatrixClient } from "matrix-js-sdk/src/matrix";
+import { ICreateClientOpts } from "matrix-js-sdk/src/matrix";
+import { ClientEvent } from "matrix-js-sdk/src/client";
+import { Visibility, Preset } from "matrix-js-sdk/src/@types/partials";
 import {
   GroupCallIntent,
   GroupCallType,
-} from "matrix-js-sdk/src/browser-index";
+} from "matrix-js-sdk/src/webrtc/groupCall";
+import { ISyncStateData, SyncState } from "matrix-js-sdk/src/sync";
+
 import IndexedDBWorker from "./IndexedDBWorker?worker";
-import Olm from "@matrix-org/olm";
-import olmWasmPath from "@matrix-org/olm/olm.wasm?url";
 
 export const defaultHomeserver =
-  import.meta.env.VITE_DEFAULT_HOMESERVER ||
+  (import.meta.env.VITE_DEFAULT_HOMESERVER as string) ??
   `${window.location.protocol}//${window.location.host}`;
 
 export const defaultHomeserverHost = new URL(defaultHomeserver).host;
 
-function waitForSync(client) {
-  return new Promise((resolve, reject) => {
-    const onSync = (state, _old, data) => {
+function waitForSync(client: MatrixClient) {
+  return new Promise<void>((resolve, reject) => {
+    const onSync = (
+      state: SyncState,
+      _old: SyncState,
+      data: ISyncStateData
+    ) => {
       if (state === "PREPARED") {
         resolve();
-        client.removeListener("sync", onSync);
+        client.removeListener(ClientEvent.Sync, onSync);
       } else if (state === "ERROR") {
         reject(data?.error);
-        client.removeListener("sync", onSync);
+        client.removeListener(ClientEvent.Sync, onSync);
       }
     };
-    client.on("sync", onSync);
+    client.on(ClientEvent.Sync, onSync);
   });
 }
 
-export async function initClient(clientOptions) {
+export async function initClient(
+  clientOptions: ICreateClientOpts
+): Promise<MatrixClient> {
   // TODO: https://gitlab.matrix.org/matrix-org/olm/-/issues/10
   window.OLM_OPTIONS = {};
   await Olm.init({ locateFile: () => olmWasmPath });
```

Comment:
GitHub won't let me comment on the `let indexedDB;` two lines below this, but can it be typed?